### PR TITLE
ComponentProxy: report differences between profile and package version

### LINKED ISSUE #48: log the components and schema version (if any)
once https://github.com/quattor/maven-tools/pull/62 is in place, we should be able to easily log versions of components (and schemas)
see also https://github.com/quattor/ncm-ncd/issues/41

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #48

### CODE REVIEW

File: src/main/perl/ComponentProxy.pm [hunk 1]
```diff
@@ -367,6 +367,33 @@ sub _setDependencies
 }
 
 
+=item _version_check
+
+Apply version related checks.
+
+Return C<SUCCESS> if there are no version-related issues;
+report an error and return undef otherwsie.
```
Comment (ned21) [inline src/main/perl/ComponentProxy.pm]: I see "version-related issues" is deliberately kept undefined, I guess pending resolution of the discussion in #41? So the doc is accurate, but potentially a little confusing ?

Comment (stdweird) [inline src/main/perl/ComponentProxy.pm]: yes, i put that in the comment below. but i could add it here, eg. `Current version only reports possible different versions, and always returns C<SUCCESS> (but behaviour might change in future version)`

Comment (jrha) [inline src/main/perl/ComponentProxy.pm]: otherwsie -> otherwise

**Comment by stdweird** [comment]
@jrha fixed typo

**Comment by jrha** [comment]
LGAFAICT

**Pull request merged**